### PR TITLE
[XPU] Add xpu support of `make triton`

### PR DESCRIPTION
**Pull request by Stonepia**
This PR is to add XPU support for `make triton`.

If a user wishes to use Triton with XPU support, the user needs to install the  [intel-xpu-backend-for-triton](https://github.com/intel/intel-xpu-backend-for-triton). 

This PR allows the user to easily install Triton for xpu backend support:

```
# clone the pytorch repo
export USE_XPU=1
make triton
```
The XPU version of triton will always be built from the source. It will cat the commit id from `.ci/docker/ci_commit_pins/triton-xpu.txt`, for example, `b8c64f64c18d8cac598b3adb355c21e7439c21de`. 

So the final call would be like:

```
pip install --force-reinstall "git+https://github.com/intel/intel-xpu-backend-for-triton@b8c64f64c18d8cac598b3adb355c21e7439c21de#subdirectory=python"
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -213,6 +213,7 @@ conda install -c pytorch magma-cuda121  # or the magma-cuda* that matches your C
 
 # (optional) If using torch.compile with inductor/triton, install the matching version of triton
 # Run from the pytorch directory after cloning
+# If you need Intel GPU support, please make sure `export USE_XPU=1`
```
Comment (EikanWang) [inline README.md:216]: Please refine the message a little bit.

**Review by EikanWang — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Stonepia) [reply README.md]: Thanks for the review! Updated in https://github.com/pytorch/pytorch/pull/126513/commits/f1728742ff2a364935362d90187c952ba53fa16b

**Review by EikanWang — APPROVED** [review]

**Comment by Stonepia** [comment]
Hello @atalman, could you help to take a look at this PR?

**Review by atalman — APPROVED** [review]
lgtm

**Review by EikanWang — APPROVED** [review]

**Pull request closed without merging**